os/arch/arm/src/imxrt/imxrt_hyperflash.c

```diff
@@ -649,6 +826,20 @@ ssize_t up_progmem_write(size_t addr, const void *buf, size_t count)
 
 	return count;
 }
+#else
+/************************************************************************************
+ * Name: up_flashinitialize
+ *
+ * Description:
+ *   Return an initialize MTD device instance.  MTD devices are not registered
+ *   in the file system, but are created as instances that can be bound to
+ *   other functions (such as a block or character driver front end).
+ *
+ ************************************************************************************/
+FAR struct mtd_dev_s *up_flashinitialize(void)
```

Comment:
This should be in mtd driver layer.

Reply:
I need to finalize your request at first.
I plan to implement it as your request